File: android_automation/mobile-tests-wdio/test/pages/otp.page.js
const { existsByText } = require('../helpers/find.helper');

class OtpPage {
    get otpInput() {
        return $('android.widget.EditText');
    }

    async isDisplayed() {
        return existsByText('মোবাইল নম্বর ভেরিফাই করুন', 'android.widget.TextView', 15000);
    }

    async enterOtp(otp) {
        await this.otpInput.waitForExist({ timeout: 15000 });
        await this.otpInput.click();
        await this.otpInput.setValue(otp);
    }

    async tapVerify() {
        try { await driver.hideKeyboard(); } catch {}
        await driver.pause(500);
        const btn = await $('android.widget.Button');
        await btn.waitForExist({ timeout: 15000 });
        await btn.click();
    }
}

module.exports = new OtpPage();

File: android_automation/mobile-tests/src/test/kotlin/shikho/automation/pages/OtpPage.kt
package shikho.automation.pages

import io.appium.java_client.android.AndroidDriver
import org.openqa.selenium.By
import org.openqa.selenium.support.ui.ExpectedConditions
import org.openqa.selenium.support.ui.WebDriverWait
import java.time.Duration

class OtpPage(private val driver: AndroidDriver) {

    private val wait = WebDriverWait(driver, Duration.ofSeconds(15))

    private val titleText = By.xpath("//android.widget.TextView[@text='মোবাইল নম্বর ভেরিফাই করুন']")
    private val otpInput = By.className("android.widget.EditText")
    private val verifyButtonContainer = By.xpath("//android.widget.TextView[@text='ভেরিফাই করুন']/parent::android.view.View")

    fun isDisplayed(): Boolean {
        return wait.until(ExpectedConditions.visibilityOfElementLocated(titleText)).isDisplayed
    }

    fun enterOtp(otp: String): OtpPage {
        val input = wait.until(ExpectedConditions.elementToBeClickable(otpInput))
        input.click()
        input.sendKeys(otp)
        return this
    }

    fun tapVerify() {
        val button = wait.until(ExpectedConditions.elementToBeClickable(verifyButtonContainer))
        button.click()
    }
}
